express-nosql/index.ts

```diff
@@ -0,0 +1,25 @@
+import dotenv from "dotenv";
+import express from "express";
+import { Express, Request, Response } from "express";
+import productRouter from "./routes/productRouter";
+import cartRouter from "./routes/cartRouter";
+import authRouter from "./routes/authRouter";
+import bodyParser = require("body-parser");
+import { connectToMongoDB } from "./config/database";
+
+dotenv.config();
+
+const app: Express = express();
+const PORT = process.env.PORT;
+const uri: string = process.env.MONGO_URI as string;
+
+connectToMongoDB(uri);
+
+app.use(bodyParser.json());
+app.use("/api/products", productRouter);
+app.use("/api/profile/cart", cartRouter);
+app.use("/api/auth", authRouter);
```

Comment:
function createApp() {
place all app related code here without db connection
}

Reply:
example of validator decorator function in express-validator
```import { validationResult, ValidationChain } from 'express-validator';
import { Request, Response } from 'express';

export function ValidateSchema(schema: ValidationChain[]) {
  return function (
    target: any,
    propertyKey: string,
    descriptor: PropertyDescriptor
  ) {
    const originalMethod = descriptor.value;

    descriptor.value = function (req: Request, res: Response) {
      Promise.all(schema.map((validator) => validator.run(req))).then(() => {
        const errors = validationResult(req);
        if (!errors.isEmpty()) {
          return res.status(400).json({
            success: false,
            errors: errors.array(),
          });
        }
        originalMethod.call(this, req, res);
      });
    };

    return descriptor;
  };
}
```

Reply:
```import { LOCALES } from '@microemarket/locales';
import { body } from 'express-validator';

const SignInValidator = [
  body('email_username')
    .exists({ checkFalsy: true })
    .withMessage(LOCALES.VALIDATORS.EMAIL_USERNAME.REQUIRED),
  body('password')
    .exists({ checkFalsy: true })
    .withMessage(LOCALES.VALIDATORS.PASSWORD.REQUIRED),
];

export default SignInValidator;
```